sklearn/semi_supervised/_label_propagation.py

```diff
@@ -255,8 +255,12 @@ def fit(self, X, y):
         self : object
             Returns the instance itself.
         """
-        self._validate_params()
-        X, y = self._validate_data(X, y)
+        X, y = self._validate_data(
+            X,
+            y,
+            accept_sparse=["csr", "csc"],
```

Comment:
`BaseEstimator._validate_data` forwards this parameter to `sklearn.utils.check_array`. This function has parameter called `accept_large_sparse` which we might want to specify the value of for this call for `BaseLabelPropagation`:

https://github.com/scikit-learn/scikit-learn/blob/550fa9c28c7193095487563ebc071271427b2bc3/sklearn/utils/validation.py#L664-L669

Could you:
 - add a test which specifically passes CSR and CSC matrices with `np.int64`-typed instead of `np.int32`-typed (the default) `indices` and `indptr` to assert the behavior of classes extending `BaseLabelPropagation` (i.e. assert if those implementation accept such inputs or if they do not)
 - specify the value of `accept_large_sparse` according to this behavior?
 
To get to `np.int64`-typed `indices` and `indptr` from `np.int32`-typed `indices` and `indptr`, this should suffice:

```python
# CSR chosen without loss of generality
X_csr.indices = X_csr.indices.astype(np.int64, copy=False) 
X_csr.indptr = X_csr.indptr.astype(np.int64, copy=False)
```

Reply:
Yes! At least, let's start with this one.